lib/elixir/lib/kernel/typespec.ex

```diff
@@ -888,6 +894,10 @@ defmodule Kernel.Typespec do
     typespec((quote do: :elixir.as_boolean(unquote(arg))), vars, caller)
   end
 
+  defp typespec({:keyword, _meta, args}, vars, caller) do
```

Comment:
Taking care of `[]` and `[arg]` separately? It would make sense. Otherwise, we could check for `length(args) in 0..1`.